src/ray/core_worker/reference_count.cc

```diff
@@ -478,11 +478,9 @@ int64_t ReferenceCounter::ReleaseLineageReferences(ReferenceTable::iterator ref)
     RAY_LOG(DEBUG) << "Releasing lineage internal for argument " << argument_id;
     arg_it->second.lineage_ref_count--;
     if (arg_it->second.ShouldDelete(lineage_pinning_enabled_)) {
-      // We only decremented the lineage ref count, so the argument value
-      // should already be released.
       RAY_CHECK(arg_it->second.on_ref_removed == nullptr);
       lineage_bytes_evicted += ReleaseLineageReferences(arg_it);
-      EraseReference(arg_it);
+      DeleteReferenceInternal(arg_it, nullptr);
```

Comment:
Should we rewrite the docstring for these 2 APIs? (EraseReference, DeleteReferenceInternal). 

From reading the docstring, it is not clear why EraseReference doesn't work but DeleteReferenceInternal does from this code

Reply:
I realize that DeleteReferenceInternal also doesn't work. It need to be 

```
ReleasePlasmaObject
EraseReference
```

I'm  not very familiar with this part so I'm going to leave it here.